components/datetime/src/timezone.rs

```diff
@@ -0,0 +1,331 @@
+        &self,
+        time_zone: &impl TimeZoneInput,
+    ) -> Option<Cow<str>> {
+        self.mz_generic_short
+            .as_ref()
+            .and_then(|metazones| time_zone.metazone_id().and_then(|mz| metazones.get(mz)))
+            .map(Clone::clone)
+    }
+
+    /// Returns the time zone in long generic non-location format as defined by the UTS-35 spec.
+    /// e.g. Pacific Time
+    /// https://unicode.org/reports/tr35/tr35-dates.html#Time_Zone_Format_Terminology
+    pub(super) fn long_generic_non_location_format(
+        &self,
+        time_zone: &impl TimeZoneInput,
+    ) -> Option<Cow<str>> {
+        self.mz_generic_long
+            .as_ref()
+            .and_then(|metazones| time_zone.metazone_id().and_then(|mz| metazones.get(mz)))
+            .map(Clone::clone)
+    }
+
+    /// Returns the time zone in short specific non-location format as defined by the UTS-35 spec.
+    /// e.g. PDT
+    /// https://unicode.org/reports/tr35/tr35-dates.html#Time_Zone_Format_Terminology
+    pub(super) fn short_specific_non_location_format(
+        &self,
+        time_zone: &impl TimeZoneInput,
+    ) -> Option<Cow<str>> {
+        self.mz_specific_short
+            .as_ref()
+            .and_then(|metazones| time_zone.metazone_id().and_then(|mz| metazones.get(mz)))
+            .and_then(|specific_names| {
+                time_zone
+                    .time_variant()
+                    .and_then(|variant| specific_names.get(variant))
+            })
+            .map(Clone::clone)
+    }
+
+    /// Returns the time zone in long specific non-location format as defined by the UTS-35 spec.
+    /// e.g. Pacific Daylight Time
+    /// https://unicode.org/reports/tr35/tr35-dates.html#Time_Zone_Format_Terminology
+    pub(super) fn long_specific_non_location_format(
+        &self,
+        time_zone: &impl TimeZoneInput,
+    ) -> Option<Cow<str>> {
+        self.mz_specific_long
+            .as_ref()
+            .and_then(|metazones| time_zone.metazone_id().and_then(|mz| metazones.get(mz)))
+            .and_then(|specific_names| {
+                time_zone
+                    .time_variant()
+                    .and_then(|variant| specific_names.get(variant))
+            })
+            .map(Clone::clone)
+    }
+
+    /// Returns the time zone in localized GMT format according to the CLDR localized hour format.
+    /// This goes explicitly against the UTS-35 spec, which specifies long or short localized
+    /// GMT formats regardless of locale.
+    ///
+    /// You can see more information about our decision to resolve this conflict here:
+    /// https://docs.google.com/document/d/16GAqaDRS6hzL8jNYjus5MglSevGBflISM-BrIS7bd4A/edit?usp=sharing
+    pub(super) fn localized_gmt_format(&self, time_zone: &impl TimeZoneInput) -> Cow<str> {
+        let gmt_offset = time_zone.gmt_offset();
+        if gmt_offset.is_zero() {
+            self.zone_formats.gmt_zero_format.clone()
+        } else {
+            Cow::Owned(
```

Comment:
Nitpick: As above, please leave a TODO for us to improve this when Zibi's Pattern thing is merged.